evap/static/ts/src/notebook.ts

```diff
@@ -0,0 +1,70 @@
+declare const bootstrap: typeof import("bootstrap");
+import { assertDefinedUnwrap } from "./utils.js";
+import { CSRF_HEADERS } from "./csrf-utils.js";
+
+const NOTEBOOK_LOCALSTORAGE_KEY = "evap_notebook_open";
+
+if (localStorage.getItem(NOTEBOOK_LOCALSTORAGE_KEY) == "true") {
+    new bootstrap.Collapse(assertDefinedUnwrap(document.getElementById("notebook")));
+    new bootstrap.Collapse(assertDefinedUnwrap(document.getElementById("notebookButton")));
+    onShowNotebook();
+}
+
+assertDefinedUnwrap(document.getElementById("notebook-save-button")).addEventListener(
+    "click",
+    function (event: MouseEvent): void {
+        event.preventDefault();
+        const form = assertDefinedUnwrap(document.getElementById("notebook-form")) as HTMLFormElement;
+        const data = new FormData(form);
+
+        const target = assertDefinedUnwrap(event.target) as HTMLButtonElement;
+        const default_label = assertDefinedUnwrap(target.getAttribute("value"));
+        const sending_label = assertDefinedUnwrap(target.getAttribute("data-label-sending"));
+        const error_label = assertDefinedUnwrap(target.getAttribute("data-label-error"));
+        const cooldown_label = assertDefinedUnwrap(target.getAttribute("data-label-cooldown"));
+        target.disabled = true;
+        target.setAttribute("value", sending_label);
+
+        fetch(form.action, {
+            body: new URLSearchParams(data as any),
+            headers: CSRF_HEADERS,
+            method: "POST",
+        }).then(response => {
+            if (response.ok) {
+                target.setAttribute("value", cooldown_label);
+                setTimeout(function (): void {
+                    target.setAttribute("value", default_label);
+                    target.disabled = false;
+                }, 2000);
+            } else {
+                target.setAttribute("value", default_label);
+                target.disabled = false;
+                alert(error_label);
+            }
+        });
```

Comment:
Something we should have seen earlier, and I just now noticed because I used `fetch()` in some other code: This doesn't actually handle a "server is not reachable" case, see https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#checking_that_the_fetch_was_successful.

If the server is not reachable, the promise will reject, so we have to `.catch()` that.

I'd propose so build it like this:
```js
fetch(form.action, {
    // fetch arguments
}).then(response => {
    assert(response.ok);
    // success handling here
}).catch(error => {
    // error handling here
};
```

@niklasmohrin agree / better ideas?